asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py

```diff
@@ -989,3 +989,76 @@ def download_cdd_data(protocol_name: str) -> pd.DataFrame:
     )
 
     return formatted_data
+
+
+def clean_result_network(network, console=None):
+    """
+    Cleans an incoming result network JSON file from some issues that might occur. Current procedures:
+    - removes edges that have DG==0.0, this happens when there is inconsistent stereo annotation in input ligands
+    such that after stereo enumeration there are duplicate ligands.
+    - cleans imbalanced complex/solvent legs, e.g. when some have failed or when stereo expansion was done unintionally
+
+    returns the loaded FreeEnergyCalculationNetwork.
+    """
+    import json
+    import sys
+    import tempfile
+
+    import numpy as np
+    import rich
+    from asapdiscovery.alchemy.schema.fec import FreeEnergyCalculationNetwork
+    from rich.padding import Padding
+
+    with open(network) as f:
+        d = json.load(f)
+        input_results = d["results"]["results"]
+
+    # 1. remove edges where DG is 0.0
+    cleaned_results = []
+    for result in input_results:
+        if not result["estimate"]["magnitude"] == 0.0:
+            cleaned_results.append(result)
+
+    num_0_0_removed = len(input_results) - len(cleaned_results)
+
+    # 2. balance between complex/solvent replicates, such that n=N=1
+    deduped_results_dict = (
+        {}
+    )  # make a dict first that we can query; if an edge has multiple results it'll be a list of results
+    for result in cleaned_results:
+        transform = f"{result['ligand_a']}~{result['ligand_b']}_{result['phase']}"
+        if transform in deduped_results_dict:
+            deduped_results_dict[transform].append(result)
+        else:
+            deduped_results_dict[transform] = [result]
```

Comment:
Use `defaultdict(list)` to avoid this pattern.  Something like the below should work. 

```suggestion
   from collections import defaultdict

    deduped_results_dict = defaultdict(list)
    for result in cleaned_results:
        transform = f"{result['ligand_a']}~{result['ligand_b']}_{result['phase']}"
        deduped_results_dict[transform].append(result)
 ```

Reply:
cheers!